utils/tests/utils.go

```diff
@@ -31,7 +31,11 @@ func StartHttpServer(handlers HttpServerHandlers) (int, error) {
 			panic(err)
 		}
 	}()
-	return listener.Addr().(*net.TCPAddr).Port, nil
+	tcpAddr, ok := listener.Addr().(*net.TCPAddr)
+	if !ok {
+		panic("Couldn't assert listener address to tcpAddr")
```

Comment:
We should avoid throwing a panic on the main thread. In contrast to the above panic on line 31, which refers to a subthread.